scalafix-core/src/main/scala/scalafix/v1/Symbol.scala

```diff
@@ -16,7 +16,7 @@ final class Symbol private (val value: String) {
   def isGlobal: Boolean = value.isGlobal
   def isLocal: Boolean = value.isLocal
   def owner: Symbol = Symbol(value.owner)
-  def displayName: String = value.desc.name.value
+  def displayName: String = encode(value.desc.name.value)
```

Comment:
maybe I am off but could https://gitter.im/scalacenter/scalafix?at=5f72c62cefefe620f2569259 be used?

Reply:
If you're inserting `Symbol.displayName()` into source code then it should be wrapped inside `scala.meta.Term.Name(...)`. This won't handle the case however when you need a leading space before the `:` colon in `def * : Int = 42`, the `ExplicitResultTypes` rule should have special logic for that to insert a the space if `!Character.isJavaIdentifierPart(lastCharacter))`.

Reply:
The motivation for modifying displayName is that is used in SemanticTree.toString. I am using it to insert valid synthetics into code source. It works except for Symbol names that are backquoted. 
Do you recommend to modify instead directly pprint for symbol to this :
```
  def pretty(sym: Symbol): Doc =
    Doc.text(scala.meta.Name(sym.displayName).toString())
``` 
Instead of  
```
  def pretty(sym: Symbol): Doc =
    Doc.text(ym.displayName)
```

Thanks